Make it possible to predefine password for SCRAM-SHA users (#5217)
strimzi/strimzi-kafka-operator: documentation/modules/con-securing-client-authentication.adoc

----
echo "Z2VuZXJhdGVkcGFzc3dvcmQ=" | base64 --decode
----

> === Providing custom password

Reviewer: ```suggestion
=== Custom password configuration
```

> When the user is created, Strimzi will normally generate a new random password.

Reviewer: ```suggestion
When a user is created, Strimzi generates a random password.
```

> In case you want to use your own password instead of the one generated by Strimzi, you can create a secret with the password and reference it in the `KafkaUser` resource:

Reviewer: ```suggestion
You can use your own password instead of the one generated by Strimzi. To do so, create a secret with the password and reference it in the `KafkaUser` resource:
```

> .An example `KafkaUser` with SCRAM-SHA-512 authentication and providing password

Reviewer: ```suggestion
.An example `KafkaUser` with a password set for SCRAM-SHA-512 authentication
```

[source,yaml,subs="attributes+"]
----
apiVersion: {KafkaUserApiVersion}
kind: KafkaUser
metadata:
  name: my-user
  labels:
    strimzi.io/cluster: my-cluster
spec:
  authentication:
    type: scram-sha-512
    password:
      valueFrom:
        secretKeyRef:
          name: my-secret <1>
          key: my-password <2>
  # ...
----

> <1> Name of the secret with your predefined password

Reviewer: ```suggestion
<1> Name of the secret providing the predefined password
```

> <2> Key under which the password is stored inside the secret

Reviewer: ```suggestion
<2> The key for the password stored inside the secret.
```
